### DISCUSSION TITLE
RadioLib LoRa Packet Structure

### DISCUSSION THREAD

**Discussion by Jaimi5** [jgromes/RadioLib#526] [Q&A] [answered]
Hi!
I am currently building a LoRaMesher with RadioLib and I would like to be sure that I understand everything right!

When enabling the CRC, and having a 4/7 CR configured, is it right the bellow image?

![PacketStructure drawio (6)](https://user-images.githubusercontent.com/33554367/169843932-865b32ae-0979-4297-8b9e-4099efcef4e5.png)
 
It is always enabled the PHDR_CRC (header CRC)? I don't fully understand which CRC I am enabling.

Hope this is not a dumb question and thanks for your time!

**Comment by jgromes** [comment, marked as answer]
To answer the question, here's a picture of the LoRa packet description from SX1262 datasheet (you didn't mention which device are you using, but the packet format is generally the same so it doesn't matter anyway):

![Screenshot_24](https://user-images.githubusercontent.com/3407192/169858609-87e5f374-31ed-4c42-beab-3787136e1671.png)

The header is only sent in explicit header mode (enabled by default), will always be sent with CR=4/8 and will always include its own CRC. This is all handled by the SX12xx device. 

The CRC enabling/disabling done by RadioLib only affects the payload CRC at the end of the packet. The coding rate you select will in turn only affect the payload, and the payload CRC. There's no coding done on the preamble (since it's just a bunch of upchirps), and the header always uses CR=4/8.

**Reply by Jaimi5** [reply]
That was all the answers I need! Thank you so much!
